realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -426,6 +427,68 @@ public SyncUser run() {
         }.start();
     }
 
+    /**
+     * Helper method for Admin users in order to look up a {@code SyncUser} using the identity provider and the used username.
+     * @param provider identity providers {@link io.realm.SyncCredentials.IdentityProvider}.
+     * @param providerId username or email used to create the account first time.
+     * @return {@code SyncUser} associated with the given identity provider and providerId.
+     * @throws ObjectServerError if the user could not be found.
```

Comment:
Not sure I follow? Wouldn't that just mean that different `RealmObjectServer` error would be returned? I was wondering if we should throw an error at all? Or do you mean that until we can correctly detect 404, we have no way of correctly knowing that was what happened?